docs/mattermost-boards-dev-guide.md

```diff
@@ -0,0 +1,62 @@
+# Mattermost Boards Developer's Guide
+
+[Mattermost Boards](https://mattermost.com/boards/) is the Focalboard (aka Boards) plugin running in [Mattermost]. It is pre-packaed, and runs out of the box with Mattermost 6.0 and later.
+
+To build your own version of it:
+1. Build [mattermost-plugin](https://github.com/mattermost/focalboard/tree/main/mattermost-plugin) in the [Focalboard repo](https://github.com/mattermost/focalboard)
+2. Upload it as a [custom plugin to your Mattermost server](https://developers.mattermost.com/integrate/admin-guide/admin-plugins-beta/#custom-plugins)
+
+Here are the steps in more detail:
+
+### Building the Boards plugin
+
+Fork or clone the [Focalboard repo](https://github.com/mattermost/focalboard), and follow the steps in the readme to set up your dev environment.
+
+Install dependencies:
+```
+# First-time setup dependencies
+cd mattermost-plugin/webapp
+npm install --no-optional
+cd ../..
+make prebuild
+```
+
+Build the plugin:
+```
+# Build webapp
+make webapp
+
+# Build plugin
+cd mattermost-plugin
+make dist
+```
+
+Refer to the [dev-release.yml](https://github.com/mattermost/focalboard/blob/main/.github/workflows/dev-release.yml#L168) workflow for the up-to-date commands that are run as part of CI.
+
+### Uploading the plugin to your server
+
+You can manually upload the plugin to your Mattermost Server:
+1. Enable [custom plugins](https://developers.mattermost.com/integrate/admin-guide/admin-plugins-beta/#custom-plugins) by setting `PluginSettings > EnableUploads` to `true` in the Mattermost `config.json` file
+2. Navigate to **System Console > Plugins > Management** and upload the generated `.tar.gz` package in your `mattermost-plugin/dist` directory
+3. Enable it (if needed)
+
+Alternatively, you can install Mattermost locally and use `make deploy` to auto-deploy it for you:
+
+First, build and run Mattermost locally:
+1. Follow the [Mattermost Developers Guide](https://developers.mattermost.com/contribute/server/developer-setup/) to set up your environment
+  * In particuler, make sure Docker is set up and running
+2. Clone / fork [mattermost-webapp](https://github.com/mattermost/mattermost-webapp) and `make build`
+3. Clone / fork [mattermost-server](https://github.com/mattermost/mattermost-server)
+4. Edit `config.json` with a site url, enable [plugin uploads](https://developers.mattermost.com/integrate/admin-guide/admin-plugins-beta/#custom-plugins), and enable [local mode](https://docs.mattermost.com/configure/configuration-settings.html#enable-local-mode)
```

Comment:
@sbishel, @wiggin77, do you know where the `config.json` file is in the mattermost-server repo? I see a `test-config.json` file, is that the one?

Reply:
@chenilim it is generated on first run.

Reply:
Thanks! I also found a [thread on this today](https://community-daily.mattermost.com/private-core/pl/n9qqdgg9ptnppfjrmcri1bze8y). So, should a new developer run `make config-reset`, edit the generated config.json, then run `make run-server`?